docs/using/components/configuration/configuration.rst

```diff
@@ -50,21 +50,44 @@ file type where the information is stored or a number of other sections.
 
 .. jsonschema:: schemas/model.yml
 
-All types of ``structure`` definitions have two keywords in common: ``v_inner_boundary`` and ``v_outer_boundary``.
+Abundances
+^^^^^^^^^^
+The ``abundance`` section has a possible ``file`` parameter with ``type`` (currently only ``artis`` is allowed)
+and a ``name`` parameter giving a path to a file containing the abundance information.
+
+.. jsonschema:: schemas/model_definitions.yml#/definitions/abundances/file
+
+.. jsonschema:: schemas/model_definitions.yml#/definitions/abundances/uniform
+
 In the ``structure`` section, one can specify a ``file`` section containing a ``type`` parameter
 (currently only ``artis`` is supported``) and a ``name`` parameter giving a path top a file.
 
 If one doesn't specify a ``file`` section, the code requires two sections (``velocities`` and ``densities``) and a
 parameter ``no_of_shells``. ``no_of_shells`` is the requested number of shells for a model. The ``velocity`` section
 requires a ``type``. Currently, only ``linear`` is supported.
 
+Structure
+^^^^^^^^^
+All types of ``structure`` definitions have two keywords in common: ``v_inner_boundary`` and ``v_outer_boundary``.
+
+.. jsonschema:: schemas/model_definitions.yml#/definitions/structure/file
+
+.. jsonschema:: schemas/model_definitions.yml#/definitions/structure/specific
+
+Densities
+"""""""""
```

Comment:
Actually, densities should be a subsection of "Specific Structure" (as you can compare my proposed TOC with your rendered TOC after these changes) - I know it seems awkward but is it possible to somehow nest it within specific structure schema (L75)'s title?

Reply:
This seems to be difficult as we are on `<H6>` at the moment on those heading. I don't think we can move it under the section 🤔 . What are your opinions, @jaladh-singhal ?

Reply:
I have it set up so that the `Density` section is under the `Structure Section` but it seems so that it is not being generated correctly 🤔

Reply:
Actually, the `jsonschema` extension is creating the appropriate headings based on the schema that I have linked. I don't think I can make it so that the Density subsection comes under the Specific Structure section because the Specific Structure is at `<H6>` & Density is also at `<H6>`.

Reply:
@DhruvSondhi I checked it in web inspector before commenting that:
- Structure is at H4
  - File Structure is at H5
  - Densities is at H5
     - Branch85_w7 Density is at H6
     - Other density types also at H6
And Overview TOC also shows same heirarchy that's why

I'm not sure how you're checking it but they are not all at H6 - I'll say to use web inspector in browser dev tools.

The actual problem here is how  `.. jsonschema::` renders and how rst format allows us to specify nesting in headings - both are leading to same heading levels. It seems to me that jsonschema when rendered, its title (i.e. Specific Structure) takes one next heading level in the section in which it is placed (i.e. Structure with H4) - so it becomes H5. And, the different underline characters in "Densities" as compared to "Structure" makes rst renderer think that Densities is subheading in Structure hence also making it H5.

That's why I said if it is possible to specify that "Densities" is subheading to jsonschema "Specific Structure"?

Reply:
Since @smithis7 will do changes with this configuration page in his doc restructure project - we can leave it for now. Hence merging - thanks for addressing other 2 comments.

Reply:
Then I think so the document I linked pertaining to the heading hierarchy in `rst` is wrong